tools/cjxl_main.cc

```diff
@@ -934,6 +938,62 @@ void ProcessFlags(const jxl::extras::Codec codec,
       });
 }
 
+struct JxlOutputProcessor {
+  bool SetOutputPath(const std::string& path) {
+    outfile.reset(new FileWrapper(path, "wb"));
+    if (!*outfile) {
+      fprintf(stderr,
+              "Could not open %s for writing\n"
+              "Error: %s",
+              path.c_str(), strerror(errno));
+      return false;
+    }
+    return true;
+  }
+
+  JxlEncoderOutputProcessor GetOutputProcessor() {
+    return JxlEncoderOutputProcessor{this, GetBuffer, ReleaseBuffer, Seek,
+                                     SetFinalizedPosition};
+  }
+
+  static void* GetBuffer(void* opaque, size_t* size) {
+    JxlOutputProcessor* self = reinterpret_cast<JxlOutputProcessor*>(opaque);
+    const size_t buffer_pos = self->position - self->finalized_position;
+    if (buffer_pos + *size > self->output.size()) {
+      self->output.resize(buffer_pos + *size, 0xDA);
+    }
+    *size = self->output.size() - buffer_pos;
+    return self->output.data() + buffer_pos;
+  }
+
+  static void ReleaseBuffer(void* opaque, size_t written_bytes) {
+    JxlOutputProcessor* self = reinterpret_cast<JxlOutputProcessor*>(opaque);
+    Seek(opaque, self->position + written_bytes);
```

Comment:
Why don't we just fwrite here and then resize the output vector to 0, getting rid of tracking finalized_position and making position match the file's position?

Reply:
Thanks, that is indeed easier, I was trying to write the file sequentially for some reason, but with fseek we don't need to. I kept finalized position so that we know the file size at the end, but position we don't need since that is implicit in the output file.

Reply:
At the end, the finalized position is equal to the last byte written, so you could SEEK_END and ftell, but possibly keeping the finalized position is easier.